runtime-common/search-index.ts

```diff
@@ -276,10 +273,13 @@ export class SearchIndex {
   }
 
   private async visitFile(url: URL) {
+    let localPath = this.realmPaths.local(url);
+    let fileRef = await this.readFileAsText(localPath);
+    if (!fileRef) {
+      return;
+    }
+    let { content, lastModified } = fileRef;
```

Comment:
Let's also make sure to test if the file specifically matches the ignore pattern and skip it if it does